### PR TITLE
fixed code quality issues

### PR DESCRIPTION
**Pull request by withshubh**
- [x]  Read Contributing.md
- [x] Solves issue #1693 

### Summary of Changes

- Replaced `strings.Index` with `strings.Contains`
- Removed unnecessary use of slice
- Combined multiple `append`s into a single call
- Added .deepsource.toml

### CODE REVIEW

File: core/errgroup/errgroup.go [hunk 1]
```diff
@@ -210,7 +210,7 @@ func (g *Group) Error() (s string) {
 }
 
 func (g *Group) getAllErrors() []error {
-	list := g.Errors[:]
+	list := g.Errors
```
Comment (AlbinoGeek) [inline core/errgroup/errgroup.go:213]: I don't think this change is equivocal.

Comment (withshubh) [reply core/errgroup/errgroup.go:213]: Hey @AlbinoGeek 

I am sure what you mean. You can have a look at the description of the issue being resolved [here](https://deepsource.io/gh/withshubh/iris/issue/CRT-A0016/description/).

**Comment by kataras** [comment]
@withshubh I've added deepsource on this repository as well, I didn't know you are part of their team, good job mate.

**Comment by kataras** [comment]
@withshubh I've added deepsource on this repository as well, I didn't know you are part of their team, good job mate. However, not all provided "fixes" are really fixes, Go support is not totally completed yet, right?

**Comment by withshubh** [comment]
@kataras Thanks :sparkling_heart: 

DeepSource Go analyzer identifies more than 200+ code quality issues and supports autofix for most commonly occurring issues. Our team actively updates all the analyzers with new rules. Learn more about Go analyzer [here](https://deepsource.io/go-static-code-analysis/).

Regarding the fixes, the first two fixes the anti-patterns while the third one boosts minor performance.

**Comment by kataras** [comment]
I've already joined to the slack @withshubh ^_^ but I don't have enough time lately to integrate with you guys yet :/ Your product looks legit and very promising though, good job!

**Comment by withshubh** [comment]
> I've already joined to the slack @withshubh ^_^ but I don't have enough time lately to integrate with you guys yet :/ Your product looks legit and very promising though, good job!

Thanks! @kataras ✨ 💖

**Pull request merged**